### PR TITLE
TINY-9964: Fixed issue with pressing space inside summary on firefox

### PR DESCRIPTION
**Pull request by spocke**
Related Ticket: TINY-9964

Description of Changes:
* Fixed issue with space key not working inside summary elements in Firefox

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/core/main/ts/keyboard/InsertSpace.ts [hunk 1]
```diff
@@ -45,7 +49,24 @@ const insertSpaceOrNbspAtSelection = (editor: Editor): Optional<() => void> => {
...
+const insertSpaceInSummaryAtSelectionOnFirefox = (editor: Editor): Optional<() => void> => {
+  const insertSpaceThunk = () => {
+    const root = SugarElement.fromDom(editor.getBody());
+
+    if (!editor.selection.isCollapsed()) {
+      editor.getDoc().execCommand('Delete');
+    }
+
+    const pos = CaretPosition.fromRangeStart(editor.selection.getRng());
+    insertSpaceOrNbspAtPosition(root, pos).each((pos) => editor.selection.setRng(pos.toRange()));
```
Comment (ltrouton) [inline modules/tinymce/src/core/main/ts/keyboard/InsertSpace.ts:62]: Wondering if we can use the `setSelection` function that `insertSpaceOrNbspAtSelection` uses
```suggestion
    insertSpaceOrNbspAtPosition(root, pos).each(setSelection);
```

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/keyboard/InsertSpace.ts (hunk 1 above)
Comment (spocke) [reply modules/tinymce/src/core/main/ts/keyboard/InsertSpace.ts]: Right good point noticed that I returned true that is never used so I changed that as well.

File: modules/tinymce/CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 - Tab navigation incorrectly stopped around `iframe` dialog component. #TINY-9815
+- It was not possible to press space to insert a space character inside a summary element on Firefox. #TINY-9954
```
Comment (vpyshnenko) [inline modules/tinymce/CHANGELOG.md:14]: Wrong ticket number

Comment (spocke) [reply modules/tinymce/CHANGELOG.md]: Whoops slip of the finger. Fixed that thanks.

**Review by vpyshnenko — APPROVED** [review]

**Pull request merged**